c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -645,37 +654,54 @@ class DeviceCachingAllocator {
     return (p.block != nullptr);
   }
 
-  bool free_cached_blocks()
+  bool release_cached_block(const AllocParams& p)
```

Comment:
This function is called in only one place so far, and a comment exists at the call site.  However, to avoid someone later trying to use this function incorrectly, it is probably worth commenting here what this function does.  Its name implies it will attempt to free any block of the given `AllocParams` size, but in reality it only tries to free the smallest possible non-split i.e. oversize block.  This is the behavior we were intending.
```suggestion
  /** Free one oversize cached block of sufficient size **/
  bool release_cached_block(const AllocParams& p)
```